### ISSUE TITLE
keyword arguments not working for C

### ISSUE THREAD

**Issue by nhamidn** [pyccel/pyccel#696]
There is a missing declaration when using keyword arguments in C

for example this python code : 
```
#$ header function f2(int, int) results(int)
def f2(x, m=None):
    if m is None:
        y = x + 1
    else:
        y = x - 1
    return y

z1 = f2(1)
z2 = f2(1, m=0)
```
generate this main C code which fail in the compilation : 

```
int main()
{
    int64_t z1;
    int64_t z2;

    z1 = f2(1, NULL);
    z2 = f2(1, &m);
    return 0;
}
```
```
error: use of
      undeclared identifier 'm'
```

After testing it appear that there is a problem with keyword arguments

This python code fail :

```
#$ header function f2(int, int) results(int)
def f2(x, m):
    if m == 0:
        y = x + 1
    else:
        y = x - 1
    return y


z1 = f2(1, m=0)
```

passing values to optional parameters working well, and it fail when combining.

Language: C

_Originally posted by @nhamidn in https://github.com/pyccel/pyccel/pull/687#discussion_r561694775_

**Comment by EmilyBourne** [comment]
Duplicate of #876

**Comment by EmilyBourne** [comment]
Blocked by #886
